# EDA by el equipo ganador

In [172]:
import os

In [173]:
from google.colab import drive
drive.mount('/gdrive')

Drive already mounted at /gdrive; to attempt to forcibly remount, call drive.mount("/gdrive", force_remount=True).


In [174]:
PATH='/gdrive/My Drive/ALGORITMOCORRECTO/'

# Initial config

In [175]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [176]:
pd.options.display.float_format = '{:,.5f}'.format

In [177]:
df=pd.read_csv(f'{PATH}/train_data.csv')

In [178]:
df.head()

id      area  ascensor  ...  tiponegocio  latitud  longitud
0  1205404  55.00000       nan  ...        Venta  4.56096 -74.09489
1  1204371  43.00000       nan  ...        Venta  4.57696 -74.10008
2  1125353 360.00000       nan  ...        Venta  4.57500 -74.10100
3  1125506  60.00000       nan  ...        Venta  4.57592 -74.10804
4  1204569  72.00000       nan  ...        Venta  4.55900 -74.09700

[5 rows x 32 columns]

In [179]:
columns=list(df.columns)
columns.pop(0)

'id'

In [180]:
#TODO: Quitar id cuando se va a entrenar el modelo
# df.pop("id")

In [181]:
def count_null_values(df: pd.DataFrame):
  for i in df.columns:
    print(f'Column: {i}')
    print('Null values:', df[i].isna().sum())

In [182]:
count_null_values(df)

Column: id
Null values: 0
Column: area
Null values: 378
Column: ascensor
Null values: 156816
Column: balcon
Null values: 184307
Column: banos
Null values: 1439
Column: banoservicio
Null values: 232257
Column: conjuntocerrado
Null values: 0
Column: cuartoservicio
Null values: 188996
Column: deposito
Null values: 137075
Column: estrato
Null values: 7682
Column: estudio
Null values: 167802
Column: garajes
Null values: 31627
Column: gimnasio
Null values: 171733
Column: habitaciones
Null values: 757
Column: halldealcobas
Null values: 179848
Column: parqueaderovisitantes
Null values: 184578
Column: piscina
Null values: 226913
Column: piso
Null values: 119388
Column: porteria
Null values: 225658
Column: remodelado
Null values: 221362
Column: saloncomunal
Null values: 135545
Column: terraza
Null values: 201781
Column: tiempodeconstruido
Null values: 60578
Column: vigilancia
Null values: 101394
Column: vista
Null values: 142356
Column: zonalavanderia
Null values: 96358
Column: valoradministraci

Segun los datos vistos que son null se procede a eliminar los datos de las siguientes columnas:
- Habitaciones
- Estratos (Ver casas cercanas)
- Area
---Revisar---Parqueadero

In [183]:
df.describe()

id                area  ...       latitud      longitud
count   244,488.00000       244,110.00000  ... 244,488.00000 244,488.00000
mean    601,739.85052        16,543.88419  ...       4.68666     -74.07231
std     405,403.87329     6,307,176.95233  ...       0.04900       0.03824
min           6.00000             0.00000  ...       3.89066     -74.36217
25%     283,131.75000            69.00000  ...       4.65798     -74.09017
50%     529,063.50000           105.00000  ...       4.69500     -74.05900
75%     905,776.25000           184.00000  ...       4.72013     -74.04547
max   6,262,945.00000 3,003,514,577.00000  ...       4.83471     -73.99017

[8 rows x 28 columns]

## Casas con estrato null

In [184]:
estrato_null = df[df.estrato.isnull()]

In [185]:
estrato_null.head()

id      area  ascensor  ...  tiponegocio  latitud  longitud
1    1204371  43.00000       nan  ...        Venta  4.57696 -74.10008
34   1204941  48.00000       nan  ...        Venta  4.62497 -74.18092
129  1073604 345.00000       nan  ...        Venta  4.73248 -74.07935
147   243819  80.00000       nan  ...        Venta  4.58100 -74.09300
148   243909  57.00000       nan  ...        Venta  4.57197 -74.13370

[5 rows x 32 columns]

In [186]:
count_null_values(estrato_null)

Column: id
Null values: 0
Column: area
Null values: 347
Column: ascensor
Null values: 4617
Column: balcon
Null values: 6796
Column: banos
Null values: 206
Column: banoservicio
Null values: 6867
Column: conjuntocerrado
Null values: 0
Column: cuartoservicio
Null values: 6340
Column: deposito
Null values: 5012
Column: estrato
Null values: 7682
Column: estudio
Null values: 5644
Column: garajes
Null values: 2006
Column: gimnasio
Null values: 6646
Column: habitaciones
Null values: 121
Column: halldealcobas
Null values: 7116
Column: parqueaderovisitantes
Null values: 4507
Column: piscina
Null values: 7065
Column: piso
Null values: 3372
Column: porteria
Null values: 4668
Column: remodelado
Null values: 6740
Column: saloncomunal
Null values: 4510
Column: terraza
Null values: 7254
Column: tiempodeconstruido
Null values: 2834
Column: vigilancia
Null values: 3510
Column: vista
Null values: 7012
Column: zonalavanderia
Null values: 3717
Column: valoradministracion
Null values: 5502
Column: valorvent

Para el dataset de entrenamiento se encontraron 378 datos null y en el subdataset donde estan los datos null de el estrato se encontraron 347 datos null

In [187]:
estrato_null.describe()

id          area  ...     latitud    longitud
count     7,682.00000   7,335.00000  ... 7,682.00000 7,682.00000
mean    377,599.25397     185.75379  ...     4.68563   -74.07644
std     273,493.88141   2,768.14700  ...     0.05219     0.03907
min           6.00000       0.00000  ...     4.28900   -74.21400
25%     207,842.75000      65.00000  ...     4.65186   -74.10193
50%     353,742.50000      97.00000  ...     4.69399   -74.06433
75%     462,862.25000     173.00000  ...     4.72313   -74.04730
max   1,405,545.00000 234,234.00000  ...     4.82420   -74.01057

[8 rows x 28 columns]

In [188]:
df[(df['latitud']>4.68) & (df['latitud']<4.69)]

id      area  ascensor  ...  tiponegocio  latitud  longitud
42      1180864  66.00000       nan  ...        Venta  4.68800 -74.06400
43      1062333  87.00000       nan  ...        Venta  4.68951 -74.09009
59       958972  90.00000       nan  ...        Venta  4.68500 -74.15379
82      1190674  58.00000       nan  ...        Venta  4.68036 -74.05491
83      1100647  37.00000       nan  ...        Venta  4.68075 -74.05686
...         ...       ...       ...  ...          ...      ...       ...
244220   263980  68.00000   1.00000  ...        Venta  4.68580 -74.04809
244221   480796  54.00000   1.00000  ...        Venta  4.68738 -74.05057
244410   551439 150.00000   1.00000  ...        Venta  4.68745 -74.04966
244413  1408248 168.00000   1.00000  ...        Venta  4.68931 -74.05256
244419   386508 126.00000   1.00000  ...        Venta  4.68832 -74.04498

[15368 rows x 32 columns]

1 grado = 111.32km

In [189]:
# import plotly.express as px

# px.set_mapbox_access_token('pk.eyJ1IjoibmVzdG9yc2dhcnpvbmMiLCJhIjoiY2s0Mzl1cThlMDVxODNqcG1oOGhjNjV3bCJ9.-IOThlJxTnp44lE0_v-HAw')
# fig = px.scatter_geo(df,
#                     lat=df.latitud,
#                     lon=df.longitud,
#                      text = df.valorventa,
                     
#                     )
# fig.show()

Estrategia: calcular el precio/area y eliminar por sectores los datos atipicos usando mediana

In [190]:
pd.options.display.float_format = '{:,.2f}'.format
parametros_numericos_2Filter = ['valorventa', 'latitud', 'longitud', 'area','habitaciones']
for x in parametros_numericos_2Filter:
  df = df[(df[x].isna() == False)]

In [191]:
count_null_values(df)

Column: id
Null values: 0
Column: area
Null values: 0
Column: ascensor
Null values: 155946
Column: balcon
Null values: 183449
Column: banos
Null values: 1129
Column: banoservicio
Null values: 231215
Column: conjuntocerrado
Null values: 0
Column: cuartoservicio
Null values: 187997
Column: deposito
Null values: 136407
Column: estrato
Null values: 7241
Column: estudio
Null values: 166714
Column: garajes
Null values: 30858
Column: gimnasio
Null values: 171103
Column: habitaciones
Null values: 0
Column: halldealcobas
Null values: 178817
Column: parqueaderovisitantes
Null values: 183826
Column: piscina
Null values: 225847
Column: piso
Null values: 118319
Column: porteria
Null values: 224576
Column: remodelado
Null values: 220301
Column: saloncomunal
Null values: 134562
Column: terraza
Null values: 200836
Column: tiempodeconstruido
Null values: 59572
Column: vigilancia
Null values: 100657
Column: vista
Null values: 141410
Column: zonalavanderia
Null values: 95587
Column: valoradministracion
N

In [192]:
def filter_by_quantile(df: pd.DataFrame, name_col: str, min = 0, max = 0.9999):
  min_val = df[name_col].quantile(min)
  max_val = df[name_col].quantile(max)
  return df[(df[name_col] >= min_val) & (df[name_col] <= max_val)]

In [193]:
def make_plots(df: pd.DataFrame, col_names: list):
  fig, axs = plt.subplots(8, figsize=(20,80))
  # plt.subplots(2,2,)
  for i in range(len(parametros_numericos)):
    axs[i].hist(df_house_price_filter[parametros_numericos[i]])
    axs[i].set_title(parametros_numericos[i])

  plt.show()

In [194]:
pd.options.display.float_format = '{:,.5f}'.format
df_vv = filter_by_quantile(df, 'valorventa', min=0.001, max = 0.99)
df_vv.describe()

id                area  ...       latitud      longitud
count   240,734.00000       240,734.00000  ... 240,734.00000 240,734.00000
mean    602,481.99312        16,767.19541  ...       4.68683     -74.07250
std     405,467.09229     6,351,248.06117  ...       0.04897       0.03829
min           6.00000             0.00000  ...       3.89066     -74.36217
25%     283,495.25000            68.81500  ...       4.65789     -74.09070
50%     531,324.00000           104.20000  ...       4.69522     -74.05914
75%     905,805.75000           180.00000  ...       4.72034     -74.04557
max   6,262,945.00000 3,003,514,577.00000  ...       4.83471     -73.99017

[8 rows x 28 columns]

In [195]:
df_a = filter_by_quantile(df_vv, 'area', min =0.003, max=0.998)
df_a.describe()

id          area  ...       latitud      longitud
count   239,529.00000 239,529.00000  ... 239,529.00000 239,529.00000
mean    602,538.90222     142.08866  ...       4.68681     -74.07249
std     405,576.21577     109.70723  ...       0.04888       0.03828
min           6.00000      16.77000  ...       3.89066     -74.36217
25%     283,483.00000      69.00000  ...       4.65794     -74.09067
50%     531,510.00000     105.00000  ...       4.69522     -74.05913
75%     905,837.00000     180.00000  ...       4.72033     -74.04557
max   6,262,945.00000   1,400.00000  ...       4.83471     -73.99017

[8 rows x 28 columns]

In [196]:
df_va = filter_by_quantile(df_a, 'valoradministracion', max = 0.9968)
df_va.describe()

id          area  ...       latitud      longitud
count   211,587.00000 211,587.00000  ... 211,587.00000 211,587.00000
mean    608,361.52130     139.05552  ...       4.68748     -74.07220
std     399,395.71307     106.53539  ...       0.04892       0.03842
min          14.00000      16.77000  ...       3.89066     -74.36217
25%     283,325.00000      68.00000  ...       4.65872     -74.08965
50%     553,612.00000     103.00000  ...       4.69588     -74.05855
75%     938,310.50000     175.00000  ...       4.72085     -74.04531
max   1,413,181.00000   1,400.00000  ...       4.83471     -73.99017

[8 rows x 28 columns]

In [197]:
df_h = filter_by_quantile(df_va, 'habitaciones', max = 0.9995)
df_h.describe()

id          area  ...       latitud      longitud
count   211,492.00000 211,492.00000  ... 211,492.00000 211,492.00000
mean    608,340.97914     138.92089  ...       4.68749     -74.07220
std     399,421.66387     106.20979  ...       0.04892       0.03842
min          14.00000      16.77000  ...       3.89066     -74.36217
25%     283,318.75000      68.00000  ...       4.65874     -74.08964
50%     553,493.50000     103.00000  ...       4.69589     -74.05853
75%     938,317.25000     175.00000  ...       4.72085     -74.04531
max   1,413,181.00000   1,400.00000  ...       4.83471     -73.99017

[8 rows x 28 columns]

In [198]:
df = df_h[(df_h.estrato >= 1) & (df_h.estrato <= 6)]
df.describe()

id          area  ...       latitud      longitud
count   209,096.00000 209,096.00000  ... 209,096.00000 209,096.00000
mean    608,822.27590     138.73543  ...       4.68751     -74.07217
std     399,267.74668     105.89647  ...       0.04885       0.03842
min          14.00000      16.77000  ...       3.89066     -74.36217
25%     283,491.75000      68.00000  ...       4.65885     -74.08958
50%     554,633.00000     102.45000  ...       4.69594     -74.05848
75%     938,270.25000     175.00000  ...       4.72086     -74.04529
max   1,413,181.00000   1,400.00000  ...       4.83471     -73.99017

[8 rows x 28 columns]

In [199]:
df['val_mcuadrado'] = df['valorventa']/df['area']

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [200]:
df.describe()

id          area  ...      longitud    val_mcuadrado
count   209,096.00000 209,096.00000  ... 209,096.00000    209,096.00000
mean    608,822.27590     138.73543  ...     -74.07217  5,088,317.88373
std     399,267.74668     105.89647  ...       0.03842  2,236,138.53057
min          14.00000      16.77000  ...     -74.36217     92,436.97479
25%     283,491.75000      68.00000  ...     -74.08958  3,620,337.99982
50%     554,633.00000     102.45000  ...     -74.05848  4,795,698.92473
75%     938,270.25000     175.00000  ...     -74.04529  6,250,000.00000
max   1,413,181.00000   1,400.00000  ...     -73.99017 99,722,222.22222

[8 rows x 29 columns]

In [201]:
df_mc = filter_by_quantile(df, 'val_mcuadrado', min = 0.0001, max = 0.9997)
df_mc.describe()


id          area  ...      longitud    val_mcuadrado
count   209,012.00000 209,012.00000  ... 209,012.00000    209,012.00000
mean    608,815.14953     138.68420  ...     -74.07217  5,078,681.68453
std     399,286.22962     105.60653  ...       0.03842  2,135,464.24111
min          14.00000      16.77000  ...     -74.36217    244,541.48472
25%     283,486.75000      68.00000  ...     -74.08957  3,620,337.99982
50%     554,585.00000     102.79500  ...     -74.05847  4,794,520.54795
75%     938,275.50000     175.00000  ...     -74.04529  6,250,000.00000
max   1,413,181.00000   1,400.00000  ...     -73.99017 21,153,846.15385

[8 rows x 29 columns]

In [202]:
df = df_mc

In [203]:
df.columns

Index(['id', 'area', 'ascensor', 'balcon', 'banos', 'banoservicio',
       'conjuntocerrado', 'cuartoservicio', 'deposito', 'estrato', 'estudio',
       'garajes', 'gimnasio', 'habitaciones', 'halldealcobas',
       'parqueaderovisitantes', 'piscina', 'piso', 'porteria', 'remodelado',
       'saloncomunal', 'terraza', 'tiempodeconstruido', 'vigilancia', 'vista',
       'zonalavanderia', 'valoradministracion', 'valorventa', 'tipoinmueble',
       'tiponegocio', 'latitud', 'longitud', 'val_mcuadrado'],
      dtype='object')

In [204]:
df_mc.head()

id      area  ascensor  ...  latitud  longitud   val_mcuadrado
0  1205404  55.00000       nan  ...  4.56096 -74.09489 3,090,000.00000
3  1125506  60.00000       nan  ...  4.57592 -74.10804 4,333,333.33333
4  1204569  72.00000       nan  ...  4.55900 -74.09700 1,736,111.11111
5  1161678 126.00000       nan  ...  4.58815 -74.10701 4,761,904.76190
6  1106444  87.00000       nan  ...  4.58068 -74.11067 2,873,563.21839

[5 rows x 33 columns]

In [205]:
df.tiempodeconstruido.unique()

array([nan, '1 a 8 años', '16 a 30 años', '9 a 15 años', 'Más de 30 años',
       'Menos de 1 año', 'Entre 10 y 20 años', 'Entre 5 y 10 años',
       'Entre 0 y 5 años', 'Más de 20 años', 'Remodelado',
       'ntre 0 y 5 años'], dtype=object)

In [206]:
for i in df.tiempodeconstruido.unique():
  print(i)
  print(len(df[df.tiempodeconstruido==i]))

nan
0
1 a 8 años
24612
16 a 30 años
28280
9 a 15 años
21477
Más de 30 años
11771
Menos de 1 año
4279
Entre 10 y 20 años
24621
Entre 5 y 10 años
13152
Entre 0 y 5 años
15488
Más de 20 años
10486
Remodelado
1388
ntre 0 y 5 años
8


- 16 a 30 años | Más de 20 años
- ntre 0 y 5 años | Entre 0 y 5 años | 1 a 8 años
- Entre 5 y 10 años
- Remodelado
- Menos de 1 año
- 9 a 15 años | Entre 10 y 20 años
- Más de 30 años

In [207]:
df.describe()

id          area  ...      longitud    val_mcuadrado
count   209,012.00000 209,012.00000  ... 209,012.00000    209,012.00000
mean    608,815.14953     138.68420  ...     -74.07217  5,078,681.68453
std     399,286.22962     105.60653  ...       0.03842  2,135,464.24111
min          14.00000      16.77000  ...     -74.36217    244,541.48472
25%     283,486.75000      68.00000  ...     -74.08957  3,620,337.99982
50%     554,585.00000     102.79500  ...     -74.05847  4,794,520.54795
75%     938,275.50000     175.00000  ...     -74.04529  6,250,000.00000
max   1,413,181.00000   1,400.00000  ...     -73.99017 21,153,846.15385

[8 rows x 29 columns]

In [208]:
one_hot_magic={
    '0-1 anos': ['Menos de 1 año'],
    '1-10 anos': ['ntre 0 y 5 años', 'Entre 0 y 5 años', '1 a 8 años', 'Entre 5 y 10 años'],
    '10-30 anos': ['9 a 15 años', 'Entre 10 y 20 años', '16 a 30 años', 'Más de 20 años'],
    '30 anos': ['Más de 30 años'],
}

In [209]:
one_hot_magic.items()

dict_items([('0-1 anos', ['Menos de 1 año']), ('1-10 anos', ['ntre 0 y 5 años', 'Entre 0 y 5 años', '1 a 8 años', 'Entre 5 y 10 años']), ('10-30 anos', ['9 a 15 años', 'Entre 10 y 20 años', '16 a 30 años', 'Más de 20 años']), ('30 anos', ['Más de 30 años'])])

In [210]:
d = []
for k, v in one_hot_magic.items():
  a = pd.Series([])
  for y in v:
    if len(df.value_counts()) <= 0:
      a=df.tiempodeconstruido == y
    else:
      a= a | df.tiempodeconstruido == y
  df[k] = a
  df[k]=df[k].replace(True, 1).replace(False, 0).astype(int)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  This is separate from the ipykernel package so we can avoid doing imports until


In [211]:
df.head()

id      area  ascensor  ...  1-10 anos  10-30 anos  30 anos
0  1205404  55.00000       nan  ...          0           0        0
3  1125506  60.00000       nan  ...          0           0        0
4  1204569  72.00000       nan  ...          0           0        0
5  1161678 126.00000       nan  ...          0           0        0
6  1106444  87.00000       nan  ...          0           0        0

[5 rows x 37 columns]

## Si no funciona (revisar filtro valor venta)

In [212]:
parametros_numericos = ['valorventa', 'latitud', 'longitud', 'valoradministracion', 'area', 'estrato', 'habitaciones']
vm = 'val_mcuadrado'



# make_plots(df_mc, parametros_numericos)

# plt.hist(df_h.habitaciones)

In [213]:
df.describe()

id          area  ...    10-30 anos       30 anos
count   209,012.00000 209,012.00000  ... 209,012.00000 209,012.00000
mean    608,815.14953     138.68420  ...       0.05017       0.05632
std     399,286.22962     105.60653  ...       0.21829       0.23053
min          14.00000      16.77000  ...       0.00000       0.00000
25%     283,486.75000      68.00000  ...       0.00000       0.00000
50%     554,585.00000     102.79500  ...       0.00000       0.00000
75%     938,275.50000     175.00000  ...       0.00000       0.00000
max   1,413,181.00000   1,400.00000  ...       1.00000       1.00000

[8 rows x 33 columns]

In [214]:
df

id      area  ascensor  ...  1-10 anos  10-30 anos  30 anos
0       1205404  55.00000       nan  ...          0           0        0
3       1125506  60.00000       nan  ...          0           0        0
4       1204569  72.00000       nan  ...          0           0        0
5       1161678 126.00000       nan  ...          0           0        0
6       1106444  87.00000       nan  ...          0           0        0
...         ...       ...       ...  ...        ...         ...      ...
244430  1393020 195.00000       nan  ...          0           0        0
244431   490880 180.00000   1.00000  ...          0           0        0
244432  1383758 183.00000       nan  ...          0           0        0
244433  1381410 108.00000   1.00000  ...          0           0        0
244434   283138 123.00000   1.00000  ...          0           0        0

[209012 rows x 37 columns]

In [215]:
count_null_values(df)

ZeroDivisionError: ignored

In [ ]:
number_of_attemps=1
df.to_csv(f'{PATH}/frankenstainV{number_of_attemps}.csv')

In [ ]:
pd.options.display.float_format = '{:,.5f}'.format
x

# Example submision

In [ ]:
df_train=pd.read_csv(f'{PATH}/train_data.csv')

In [ ]:
df_train['m2']=df_train.valorventa/df_train.area

In [ ]:
df_train[df_train.valorventa==123456789]

In [ ]:
df[df.valorventa==123456789]

In [ ]:
df_train.describe()

In [ ]:
df_example=pd.read_csv(f'{PATH}/example_submission.csv')

In [ ]:
df_example.tail()

In [ ]:
df_train.head()

In [ ]:
for i in df_example['id']:
  print(df_train[df_train['id']==i])

In [ ]:
fr = pd.read_csv(f'{PATH}frankenstainV1Estrato.csv')

In [216]:
fr.head()

field_1       id      area  ...  RESPONSABL  SHAPE_Leng   SHAPE_Area
0        3  1125506  60.00000  ...         560   455.44384 11,733.28557
1        4  1204569  72.00000  ...         560   526.72535 13,106.71596
2        8   958538  62.00000  ...         560   907.39527 43,417.20083
3       12  1084087  34.00000  ...         560   117.39756    832.53782
4       13  1204653 280.00000  ...         560   227.99827  2,922.62256

[5 rows x 51 columns]

In [217]:
fr3 = fr[fr['estrato'] == fr['ESTRATO_2']]

field_1       id      area  ...  RESPONSABL  SHAPE_Leng   SHAPE_Area
0             3  1125506  60.00000  ...         560   455.44384 11,733.28557
2             8   958538  62.00000  ...         560   907.39527 43,417.20083
4            13  1204653 280.00000  ...         560   227.99827  2,922.62256
7            22  1143323  37.00000  ...         560   554.28952 16,118.53701
9            24   985172  70.00000  ...         560   250.35338  3,542.58502
...         ...      ...       ...  ...         ...         ...          ...
136235   244425  1403961 235.50000  ...         560   510.65670 14,347.74848
136236   244430  1393020 195.00000  ...         560   240.00359  2,699.82037
136237   244431   490880 180.00000  ...         560   538.26785 16,035.20752
136238   244432  1383758 183.00000  ...         560   755.33828 28,571.90781
136239   244433  1381410 108.00000  ...         560   941.93076 22,603.30703

[101023 rows x 51 columns]

In [218]:
number_of_attemps=2
fr.to_csv(f'{PATH}/frankenstainV{number_of_attemps}.csv')